profile.d/80-systemd-osc-context.sh

```diff
@@ -50,10 +50,14 @@ __systemd_osc_context_precmdline() {
     read -r systemd_osc_context_cmd_id </proc/sys/kernel/random/uuid
 }
 
-if [[ -n "${BASH_VERSION:-}" ]]; then
-    # Whenever a new prompt is shown close the previous command, and prepare new command
-    PROMPT_COMMAND+=(__systemd_osc_context_precmdline)
+if [ -n "${BASH_VERSION:-}" ]; then
+    # Only newer bash versions (>= 5.1) support PROMPT_COMMAND as an array.
+    if [[ "$(declare -p PROMPT_COMMAND 2>&1)" =~ "declare -a" ]]; then
 
-    # PS0 is shown right after a prompt completed, but before the command is executed
-    PS0='\033]8003;start=$systemd_osc_context_cmd_id$(__systemd_osc_context_common);type=command;cwd=$(__systemd_osc_context_escape "$PWD")\033\\'"${PS0:-}"
+        # Whenever a new prompt is shown, close the previous command, and prepare new command
+        PROMPT_COMMAND+=(__systemd_osc_context_precmdline)
```

Comment:
```
/var/tmp/autopkgtest-lxc.67c4ogzr/downtmp/wrapper.sh: 58: /etc/profile.d/80-systemd-osc-context.sh: Syntax error: word unexpected (expecting ")")
```

Reply:
Hmm, this is mostly for the CI failure.
Looking at the wrapper.sh, it has `#!/bin/sh`, and I guess it does not run with bash.

But, e.g. vte-profile provides something similar code.
https://github.com/GNOME/vte/blob/5bc469790151ed30712e4b5a03c59e8d077c96c6/src/vte.sh.in#L74-L88

What does the wrapper script do? How can we avoid the check??

Reply:
It cannot be avoided if this profile.d is to be installed, as it will run by default everywhere. So it needs to work when ran from something other than bash.

Reply:
https://salsa.debian.org/gnome-team/vte2.91
Why it passes the test??

Reply:
It may not be test. I meant something around wrapper.sh.